lib/household/view/edit_household.dart

```diff
@@ -0,0 +1,185 @@
+                                      final position = await getPosition();
+                                      if (!mounted) {
+                                        return;
+                                      }
+                                      ScaffoldMessenger.of(context)
+                                          .hideCurrentSnackBar();
+                                      formKey
+                                          .currentState!.fields['geoLocation']!
+                                          .didChange(
+                                              '${position.latitude}, ${position.longitude}');
+                                    } catch (e) {
+                                      ScaffoldMessenger.of(context)
+                                        ..hideCurrentSnackBar()
+                                        ..showSnackBar(SnackBar(
+                                          content: Text(
+                                              'Could not get location: $e'),
+                                        ));
+                                    }
+                                  },
+                                  icon: const Icon(Icons.refresh))),
+                          onChanged: (value) => changed = true,
+                          readOnly: true,
+                          validator: FormBuilderValidators.required(),
+                          initialValue: state.household!.geoLocation,
+                        ),
+                        FormBuilderDateTimePicker(
+                          name: 'sensitizationDate',
+                          decoration: const InputDecoration(
+                            label: Text('Sensitization Date'),
+                            icon: Icon(Icons.calendar_today),
+                          ),
+                          onChanged: (value) => changed = true,
+                          inputType: InputType.date,
+                          lastDate: DateTime.now(),
+                          validator: FormBuilderValidators.required(),
+                          format: DateFormat('yyyy-MM-dd'),
+                          initialValue: state.household!.sensitizationDate,
+                        ),
+                        FormBuilderTextField(
+                          name: 'comments',
+                          decoration: const InputDecoration(
+                              label: Text('Comments'),
+                              icon: Icon(Icons.comment)),
+                          onChanged: (value) => changed = true,
+                          minLines: 1,
+                          maxLines: null,
+                          initialValue: state.household!.comments,
+                        ),
+                      ]),
+                    )
+                  : const Center(child: CircularProgressIndicator()),
+            ),
+          ),
+        );
+      },
+    );
+  }
+}
+
+Future<Position> getPosition() async {
```

Comment:
can we extract this to a separate class so it isn't repeated as I believe it is also used somewhere else

Reply:
Agreed, but let's do this in another MR just to keep the number of changes in this one down